GVFS/GVFS.Platform.POSIX/POSIXFileSystem.cs

```diff
@@ -65,138 +51,9 @@ public bool TryCreateDirectoryWithAdminAndUserModifyPermissions(string directory
-            private static readonly ushort IFSOCK = Convert.ToUInt16("0140000", 8);
-
-            // #define S_IXUSR     0000100     /* [XSI] X for owner */
-            private static readonly ushort IXUSR = Convert.ToUInt16("100", 8);
-
-            // #define S_IXGRP     0000010     /* [XSI] X for group */
-            private static readonly ushort IXGRP = Convert.ToUInt16("10", 8);
-
-            // #define S_IXOTH     0000001     /* [XSI] X for other */
-            private static readonly ushort IXOTH = Convert.ToUInt16("1", 8);
-
-            public static bool IsSock(ushort mode)
-            {
-                // #define  S_ISSOCK(m) (((m) & S_IFMT) == S_IFSOCK)    /* socket */
-                return (mode & IFMT) == IFSOCK;
-            }
-
-            public static bool IsExecutable(ushort mode)
-            {
-                return (mode & (IXUSR | IXGRP | IXOTH)) != 0;
-            }
-
-            [DllImport("libc", EntryPoint = "stat$INODE64", SetLastError = true)]
-            public static extern int Stat(string path, [Out] out StatBuffer statBuffer);
-
-            [StructLayout(LayoutKind.Sequential)]
-            public struct TimeSpec
-            {
-                public long Sec;
-                public long Nsec;
-            }
-
-            [StructLayout(LayoutKind.Sequential)]
-            public struct StatBuffer
-            {
-                public int Dev;              /* ID of device containing file */
-                public ushort Mode;          /* Mode of file (see below) */
-                public ushort NLink;         /* Number of hard links */
-                public ulong Ino;            /* File serial number */
-                public uint UID;             /* User ID of the file */
-                public uint GID;             /* Group ID of the file */
-                public int RDev;             /* Device ID */
-
-                public TimeSpec ATimespec;     /* time of last access */
-                public TimeSpec MTimespec;     /* time of last data modification */
-                public TimeSpec CTimespec;     /* time of last status change */
-                public TimeSpec BirthTimespec; /* time of file creation(birth) */
-
-                public long Size;          /* file size, in bytes */
-                public long Blocks;        /* blocks allocated for file */
-                public int BlkSize;        /* optimal blocksize for I/O */
-                public uint Glags;         /* user defined flags for file */
-                public uint Gen;           /* file generation number */
-                public int LSpare;         /* RESERVED: DO NOT USE! */
-
-                [MarshalAs(UnmanagedType.ByValArray, SizeConst = 2)]
-                public long[] QSpare;     /* RESERVED: DO NOT USE! */
-            }
-        }
-
-        private class NativeFileReader
-        {
-            public const int ReadOnly = 0x0000;
-            public const int WriteOnly = 0x0001;

```

Comment:
I noticed that `WriteOnly` and `Create` were dropped, but they were not used. @wilbaker @nickgra do you know why they were there at all?

Reply:
As a bit of context, these are mappings of the libc API flags `O_WRONLY` and `O_CREAT`, as defined in, e.g., FreeBSD's [`fcntl.h`](https://github.com/freebsd/freebsd/blob/def45a363e51d353f063c8876ad00ddac259c019/sys/sys/fcntl.h#L76).

It looks to me as though there was some [refactoring of native filesystem methods](https://github.com/microsoft/VFSForGit/commit/dc13c08b942e908b481ccdcf1bfe3b3736a4bfe4#diff-a279408f959e444a7a8665ecae6e4dc7) back in December which resulted in the creation of [`FastFetch.NativeUnixMethods`](https://github.com/microsoft/VFSForGit/blob/29fd6a1746fabedea35ac920ef0a2f844aeacb3e/GVFS/FastFetch/NativeUnixMethods.cs#L9) as well as `GVFS.Platform.Mac.MacFileSystem`, which was then [migrated to `POSIXFileSystem`](https://github.com/microsoft/VFSForGit/commit/262772e61be3eb0513749fa49a25ed3186643f0d#diff-4d1bba707d97f523c1c9d2f1bc15dcd8) by @kivikakk recently so we could share the common POSIX bits with Linux.

My hunch would be that because some of these flags are used in the [`FastFetch.NativeUnixMethods`](https://github.com/microsoft/VFSForGit/blob/29fd6a1746fabedea35ac920ef0a2f844aeacb3e/GVFS/FastFetch/NativeUnixMethods.cs#L35) code, they were included here by accident.

It's a good question, because it points out that if we want to use `FastFetch` on Linux, we'll need to refactor `FastFetch.NativeUnixMethods`, which is currently going to work on BSD/Darwin only as these flags like `O_CREAT` have different values on Linux, and the fields of `struct stat` are also very different.  I've created [an issue](github/VFSForGit#17) in our backlog for that!